### ISSUE TITLE
[tool_crash] IndexError: RangeError (index): Index out of range: index should be less than 267: 267

### ISSUE THREAD

**Issue by paul-soundprj** [flutter/flutter#76029]
## Command
```
flutter --no-color --no-version-check precache
```

## Steps to Reproduce
1. ...
2. ...
3. ...

## Logs
IndexError: RangeError (index): Index out of range: index should be less than 267: 267
```
#0      Uint8List.[]= (dart:typed_data-patch/typed_data_patch.dart:2225:7)
#1      Inflate._decode (package:archive/src/zlib/inflate.dart:317:20)
#2      Inflate._parseDynamicHuffmanBlock (package:archive/src/zlib/inflate.dart:231:9)
#3      Inflate._parseBlock (package:archive/src/zlib/inflate.dart:117:9)
#4      Inflate._inflate (package:archive/src/zlib/inflate.dart:87:20)
#5      new Inflate.buffer (package:archive/src/zlib/inflate.dart:19:5)
#6      ArchiveFile.decompress (package:archive/src/archive_file.dart:71:28)
#7      ArchiveFile.content (package:archive/src/archive_file.dart:62:7)
#8      _WindowsUtils._unpackArchive (package:flutter_tools/src/base/os.dart:405:45)
#9      _WindowsUtils.unzip (package:flutter_tools/src/base/os.dart:380:5)
#10     ArtifactUpdater._downloadArchive (package:flutter_tools/src/cache.dart:1512:18)
#11     _rootRunUnary (dart:async/zone.dart:1198:47)
#12     _CustomZone.runUnary (dart:async/zone.dart:1100:19)
#13     _FutureListener.handleValue (dart:async/future_impl.dart:143:18)
#14     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:696:45)
#15     Future._propagateToListeners (dart:async/future_impl.dart:725:32)
#16     Future._completeWithValue (dart:async/future_impl.dart:529:5)
#17     Future._asyncCompleteWithValue.<anonymous closure> (dart:async/future_impl.dart:567:7)
#18     _rootRun (dart:async/zone.dart:1190:13)
#19     _CustomZone.run (dart:async/zone.dart:1093:19)
#20     _CustomZone.runGuarded (dart:async/zone.dart:997:7)
#21     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:1037:23)
#22     _microtaskLoop (dart:async/schedule_microtask.dart:41:21)
#23     _startMicrotaskLoop (dart:async/schedule_microtask.dart:50:5)
#24     _runPendingImmediateCallback (dart:isolate-patch/isolate_patch.dart:118:13)
```
```
[✓] Flutter (Channel stable, 1.22.6, on Microsoft Windows [Version 10.0.18363.1016], locale ja-JP)
    • Flutter version 1.22.6 at C:\flutter
    • Framework revision 9b2d32b605 (3 weeks ago), 2021-01-22 14:36:39 -0800
    • Engine revision 2f0af37152
    • Dart version 2.10.5

[!] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at C:\android\sdk
    • Platform android-30, build-tools 30.0.3
    • ANDROID_SDK_ROOT = C:\android\sdk
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
    ✗ Android licenses not accepted.  To resolve this, run: flutter doctor --android-licenses

[!] Android Studio (version 4.1.0)
    • Android Studio at C:\Program Files\Android\Android Studio
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[!] Connected device
    ! No devices available

! Doctor found issues in 3 categories.

```

## Flutter Application Metadata
No pubspec in working directory.

**Comment by pedromassangocode** [comment]
Hi @paul-soundprj 
Do you have any reproducible steps that reproduces this issue?

**Comment by paul-soundprj** [comment]
Hi @pedromassangocode

I am not good at English, so I will translate it.
I don't know if it can be reproduced because it happened only once.

Environment:
OS: Windows 10 Pro Japanese
Memory: 8GB
SSD: 128GB

Flutter's 1.22.6 and 1.22.5 have a small capacity and an error will occur in Android Studio, so download 1.22.4 and download
At the command prompt
> flutter upgrade
So it happened.
However, the version was updated to 1.22.6.

**Comment by christopherfujino** [comment]
@paul-soundprj can you try upgrading to the newest stable version of flutter? This version (1.22.6) does not have checksum validation logic (https://github.com/flutter/flutter/commit/ac751cc50cc28d329f4b7581bbadd960918a9b64), and I suspect what happened was that the zip file that you downloaded was corrupted.

**Comment by darshankawar** [comment]
Considering the age of this issue and lack of any activity after above comment, I am going ahead and closing this as stale.
@paul-soundprj If you see same issue using latest stable version, please open new issue.
